streams/src/test/java/org/apache/kafka/streams/integration/EOSUncleanShutdownIntegrationTest.java

```diff
@@ -0,0 +1,168 @@
+
+        STREAMS_CONFIG.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, EXACTLY_ONCE);
+        STREAMS_CONFIG.put(StreamsConfig.STATE_DIR_CONFIG, TEST_FOLDER.getRoot().getPath());
+    }
+
+    @Test
+    public void shouldWorkWithUncleanShutdownWipeOutStateStore() throws InterruptedException {
+        final String appId = getClass().getSimpleName().toLowerCase(Locale.getDefault()) + "-test";
+        STREAMS_CONFIG.put(StreamsConfig.APPLICATION_ID_CONFIG, appId);
+
+        final String input = "input-topic";
+        cleanStateBeforeTest(CLUSTER, input);
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final KStream<String, String> inputStream = builder.stream(input);
+
+        final AtomicInteger recordCount = new AtomicInteger(0);
+
+        final KTable<String, String> valueCounts = inputStream
+                                                       .groupByKey()
+                                                       .aggregate(() -> "()", (key, value, aggregate) -> aggregate + ",(" + key + ": " + value + ")", Materialized.as("aggregated_value"));
+        valueCounts.toStream().peek((key, value) -> {
+            if (recordCount.incrementAndGet() >= fatalExceptionThreshold) {
+                throw new IllegalStateException("Crash on the " + fatalExceptionThreshold + " record");
+            }
+        });
+
+        final Properties producerConfig = mkProperties(mkMap(
+            mkEntry(ProducerConfig.CLIENT_ID_CONFIG, "anything"),
+            mkEntry(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, ((Serializer<String>) STRING_SERIALIZER).getClass().getName()),
+            mkEntry(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ((Serializer<String>) STRING_SERIALIZER).getClass().getName()),
+            mkEntry(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers())
+        ));
+        final KafkaStreams driver = IntegrationTestUtils.getStartedStreams(STREAMS_CONFIG, builder, true);
+
+        final File stateDir = new File(
+            String.join("/", TEST_FOLDER.getRoot().getPath(), appId, "0_0"));
+        try {
+
+            IntegrationTestUtils.produceSynchronously(producerConfig, false, input, Optional.empty(),
+                singletonList(new KeyValueTimestamp<>("k1", "v1", 0L)));
+
+            TestUtils.waitForCondition(stateDir::exists,
+                "Failed awaiting CreateTopics first request failure");
+
+            IntegrationTestUtils.produceSynchronously(producerConfig, false, input, Optional.empty(),
+                asList(new KeyValueTimestamp<>("k2", "v2", 1L),
+                    new KeyValueTimestamp<>("k3", "v3", 2L)));
+
+            TestUtils.waitForCondition(() -> recordCount.get() == RECORD_TOTAL,
+                "Expected " + RECORD_TOTAL + " records processed but only got " + recordCount.get());
+        } finally {
+            driver.close();
+            if (fatalExceptionThreshold > RECORD_TOTAL) {
+                assertTrue(stateDir.exists());
```

Comment:
Why do we want to validate this case, where no exception would be thrown, and hence no handling logic should be triggered either?

Reply:
The purpose is just for more comprehensive checking to make sure the current codepath won't wipe out states if we close clean. Let me update the test name to reflect this.

Reply:
Actually let me just simplify the test case